From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: djmips <david.c.galloway@gmail.com>
Date: Sun, 14 Jun 2026 02:34:42 -0700
Subject: [PATCH] 9276: MC6800/6809: fix BGT/LBGT branch condition (uses Carry
 instead of Zero)

GP-0: MC6800/6809: fix BGT/LBGT branch condition (uses Carry
 instead of Zero)
---
 Ghidra/Processors/MC6800/data/languages/6x09.sinc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Ghidra/Processors/MC6800/data/languages/6x09.sinc b/Ghidra/Processors/MC6800/data/languages/6x09.sinc
index d7c7e9b0b2..5ba1f91e08 100644
--- a/Ghidra/Processors/MC6800/data/languages/6x09.sinc
+++ b/Ghidra/Processors/MC6800/data/languages/6x09.sinc
@@ -1007,7 +1007,7 @@ macro PushEntireState()
 
 :BGT REL    is op=0x2E; REL
 {
-        if (($(N) == $(V)) & $(C)) goto REL;
+        if (($(N) == $(V)) & ($(Z) == 0)) goto REL;
 }
 
 :BLE REL     is op=0x2F; REL
@@ -1346,7 +1346,7 @@ macro PushEntireState()
 
 :LBGT REL2    is PAGE2; op=0x2E; REL2
 {
-        if (($(N) == $(V)) & $(C)) goto REL2;
+        if (($(N) == $(V)) & ($(Z) == 0)) goto REL2;
 }
 
 :LBLE REL2     is PAGE2; op=0x2F; REL2
-- 
2.45.1

